appserver/concurrent/concurrent-impl/src/main/java/org/glassfish/concurrent/runtime/ContextSetupProviderImpl.java

```diff
@@ -235,29 +243,29 @@ private ComponentInvocation getSavedInvocation() {
             return null;
         }
         if (setup.isClear(JNDI)) {
-            return new ComponentInvocation();
-        }
-        if (setup.isPropagated(JNDI)) {
-            return cloneComponentInvocation(currentInvocation);
+            return new ComponentInvocation(null, null, null, currentInvocation.getAppName(), null);
         }
-        return null;
+        return cloneComponentInvocation(currentInvocation);
     }
 
 
     /**
      * Check whether the application component submitting the task is still running.
      * Throw IllegalStateException if not.
      */
-    private boolean isApplicationEnabled(String appId) {
+    private void verifyApplicationEnabled(String appId) {
```

Comment:
I still think that this method brings and always brought some fragility. It is completely heuristic, while we should rather use some lifecycle stages of the application to check where we are and what we can do .... sounds like sci-fi now :-)
I think there are also other places feeling this pain in GF.

Reply:
And second question: `appId == null` are illegal always or found some legal cases where it may be `null`?

Reply:
@OndroMih, we already it the #24544 :)

Reply:
Sorry, I’m on phone, I meant this PR: https://github.com/eclipse-ee4j/glassfish/pull/24540